lib/heex_formatter/formatter.ex

```diff
@@ -17,42 +29,30 @@ defmodule HeexFormatter.Phases.Format do
   The following string will be returned:
 
   "<section>\n  <div>\n    <h1>\n      Hello\n    </h1>\n  </div>\n</section>\n"
-
-  Notice that this string is formatted. So this is supposed to be the last
-  step before writing it to a file.
   """
-
-  # Use 2 spaces for a tab
-  @tab "  "
-
-  @preformatted_tags ~w(script code style pre)
-
-  # Line length of opening tags before splitting attributes onto their own line
-  @default_line_length 98
-
-  @spec run(list(), Keyword.t()) :: String.t()
-  def run(tokens, opts) do
+  def format(tokens, opts) do
     initial_state = %{
-      html: "",
+      buffer: [],
       previous_token: nil,
       indentation: 0,
       line_length: opts[:heex_line_length] || opts[:line_length] || @default_line_length,
-      formatter_opts: opts
+      formatter_opts: opts,
+      mode: :normal
     }
 
-    result =
-      Enum.reduce(tokens, initial_state, fn token, state ->
-        new_state = token_to_string(token, state)
-
-        # Set the previous token so we can check it to know how the current tag should
-        # be formatted.
-        %{new_state | previous_token: token}
-      end)
+    tokens
+    |> Enum.reduce(initial_state, fn token, state ->
+      new_state = token_to_string(token, state)
+      %{new_state | previous_token: token}
```

Comment:
This is bothering me a bit because I guess I could store all tokens and pop the first to use as previous_token. But this also works and right now my current logic doesn't depend on a list of tokens but only the previous one. So I guess it is fine? 🤔